src/rust/catnip/runtime/mod.rs

```diff
@@ -438,22 +435,25 @@ impl NetworkRuntime for SharedDPDKRuntime {
                 assert!(header_size + body.len() >= MIN_PAYLOAD_SIZE);
 
                 // We're only using the header_mbuf for, well, the header.
-                header_mbuf.trim(header_mbuf.len() - header_size).unwrap();
+                header_mbuf.trim(header_mbuf.len() - header_size)?;
 
                 // Get the body mbuf.
                 let body_mbuf: *mut rte_mbuf = if body.is_dpdk_allocated() {
                     // The body is already stored in an MBuf, just extract it from the DemiBuffer.
-                    expect_some!(body.into_mbuf(), "'body' should be DPDK-allocated")
+                    match body.into_mbuf() {
+                        Some(body) => body,
+                        _ => return Err(Fail::new(libc::EINVAL, "'body' should be DPDK-allocated")),
+                    }
                 } else {
                     // The body is not dpdk-allocated, allocate a DPDKBuffer and copy the body into it.
-                    let mut mbuf: DemiBuffer = match self.mm.alloc_body_mbuf() {
-                        Ok(mbuf) => mbuf,
-                        Err(e) => panic!("failed to allocate body mbuf: {:?}", e.cause),
-                    };
+                    let mut mbuf: DemiBuffer = self.mm.alloc_body_mbuf()?;
                     assert!(mbuf.len() >= body.len());
                     mbuf[..body.len()].copy_from_slice(&body[..]);
-                    mbuf.trim(mbuf.len() - body.len()).unwrap();
-                    expect_some!(mbuf.into_mbuf(), "mbuf should not be empty")
+                    mbuf.trim(mbuf.len() - body.len())?;
+                    match mbuf.into_mbuf() {
+                        Some(body) => body,
+                        _ => return Err(Fail::new(libc::EINVAL, "body mbuf could not be allocated")),
```

Comment:
This error happened not because an invalid argument was supplied to the function, but rather because we could not allocate a buffers to hold the body of the pakcet. My suggestion is that we should return either  `EAGAIN` or `ENOMEM` instead.